pkg/engine/internal/planner/logical/planner_test.go

```diff
@@ -150,12 +151,48 @@ func TestConvertAST_MetricQuery_Success(t *testing.T) {
 RETURN %13
 `
 
-	require.Equal(t, expected, logicalPlan.String())
+		require.Equal(t, expected, logicalPlan.String())
 
-	var sb strings.Builder
-	PrintTree(&sb, logicalPlan.Value())
+		var sb strings.Builder
+		PrintTree(&sb, logicalPlan.Value())
 
-	t.Logf("\n%s\n", sb.String())
+		t.Logf("\n%s\n", sb.String())
+	})
+
+	t.Run(`sum by (level) (count_over_time({cluster="prod", namespace=~"loki-.*"}[5m]) / 300)`, func(t *testing.T) {
+		q := &query{
+			statement: `sum by (level) (count_over_time({cluster="prod", namespace=~"loki-.*"}[5m]) / 300)`,
+			start:     3600,
+			end:       7200,
+			interval:  5 * time.Minute,
+		}
+
+		logicalPlan, err := BuildPlan(q)
+		require.NoError(t, err)
+		t.Logf("\n%s\n", logicalPlan.String())
+
+		expected := `%1 = EQ label.cluster "prod"
+%2 = MATCH_RE label.namespace "loki-.*"
+%3 = AND %1 %2
+%4 = MAKETABLE [selector=%3, predicates=[], shard=0_of_1]
+%5 = GTE builtin.timestamp 1970-01-01T00:55:00Z
+%6 = SELECT %4 [predicate=%5]
+%7 = LT builtin.timestamp 1970-01-01T02:00:00Z
+%8 = SELECT %6 [predicate=%7]
+%9 = RANGE_AGGREGATION %8 [operation=count, start_ts=1970-01-01T01:00:00Z, end_ts=1970-01-01T02:00:00Z, step=0s, range=5m0s]
+%10 = DIV %9 300
```

Comment:
nit: I wonder whether we should change the string representation of a literal to something like


```suggestion
%10 = DIV %9 LITERAL(300)
```

or


```suggestion
%10 = DIV %9 INT64(300)
```